### PR TITLE
[-] Project: norms

### PR DESCRIPTION
**Pull request by PrestaEdit**

### CODE REVIEW

**Comment by PrestaEdit** [comment]
Hi @kpodemski  ; I correct this one :)

**Comment by PrestaEdit** [comment]
At this point, it's about 899 notices and 1534 errors (on validator) corrected.

File: classes/AdminTab.php [hunk 1]
```diff
@@ -1265,22 +1264,22 @@ public function getList($id_lang, $orderBy = NULL, $orderWay = NULL, $start = 0,
 		if (empty($orderBy))
 			$orderBy = $this->context->cookie->__get($this->table.'Orderby') ? $this->context->cookie->__get($this->table.'Orderby') : $this->_defaultOrderBy;
 		if (empty($orderWay))
-			$orderWay = $this->context->cookie->__get($this->table.'Orderway') ? $this->context->cookie->__get($this->table.'Orderway') : 'ASC';
+			$orderWay = $this->context->cookie->__get($this->table.'Orderway') ? $this->context->cookie->__get($this->table.'Orderway') : 'asC';
```
Comment (gRoussac) [inline classes/AdminTab.php]: Hi,

Can you revert to lowercase here ?

Thanks

**Comment by PrestaEdit** [comment]
Hi @gRoussac ,

I revert ASC and __CLASS__ case. :)

File: classes/controller/Controller.php [hunk 2]
```diff
@@ -502,7 +502,11 @@ protected function smartyOutputContent($content)
 			));
 
 			$javascript = $this->context->smarty->fetch(_PS_ALL_THEMES_DIR_.'javascript.tpl');
-			echo ($defer ? $html.$javascript : preg_replace('/(?<!\$)'.$js_tag.'/', $javascript, $html)).$live_edit_content.((!isset($this->ajax) || ! $this->ajax) ? '</body></html>' : '');
+
+			if ($defer)
+				echo $html.$javascript
```
Comment (jeromenadaud) [inline classes/controller/Controller.php]: Missing ;

File: classes/controller/Controller.php [hunk 3]
```diff
@@ -502,7 +502,11 @@ protected function smartyOutputContent($content)
...
+			else
+				echo preg_replace('/(?<!\$)'.$js_tag.'/', $javascript, $html)).$live_edit_content.((!isset($this->ajax) || ! $this->ajax) ? '</body></html>' : '');
```
Comment (jeromenadaud) [inline classes/controller/Controller.php]: Duplicate Brace

File: classes/order/Order.php [hunk 4]
```diff
@@ -814,17 +814,17 @@ public function hasProductReturned()
 
 	public function hasBeenPaid()
 	{
-		return count($this->getHistory((int)($this->id_lang), false, false, OrderState::FLAG_PAID));
+		return count($this->getHistory((int)$this->id_lan), false, false, OrderState::FLAG_PAID));
```
Comment (jeromenadaud) [inline classes/order/Order.php]: 'g' dropped for brace

File: classes/controller/Controller.php (hunk 3 above)
Comment (PrestaEdit) [inline classes/controller/Controller.php]: Duplicate ? Wich one ?

File: classes/controller/Controller.php (hunk 2 above)
Comment (PrestaEdit) [inline classes/controller/Controller.php]: Done.

File: classes/order/Order.php (hunk 4 above)
Comment (PrestaEdit) [inline classes/order/Order.php]: Done.

**Comment by PrestaEdit** [comment]
Hi @jeromenadaud ,

Don't know if you see it, I'd correct what you commented.

File: classes/AdminTab.php [hunk 5]
```diff
@@ -1429,19 +1428,19 @@ public function displayListHeader($token = NULL)
...
+				var token = \''.($token != null ? $token : $this->token).'\';
 				var come_from = \''.$this->table.'\';
 				var alternate = \''.($this->_orderWay == 'DESC' ? '1' : '0' ).'\';
 			</script>
 			<script type="text/javascript" src="../js/admin/dnd.js"></script>
 			';
 		}
-		echo '<table'.(array_key_exists($this->identifier,$this->identifiersDnd) ? ' id="'.(((int)(Tools::getValue($this->identifiersDnd[$this->identifier], 1))) ? substr($this->identifier,3,strlen($this->identifier)) : '').'"' : '' ).' class="table'.((array_key_exists($this->identifier,$this->identifiersDnd) && ($this->_orderBy != 'position 'AND $this->_orderWay != 'DESC')) ? ' tableDnD'  : '' ).'" cellpadding="0" cellspacing="0">
+		echo '<table'.(array_key_exists($this->identifier, $this->identifiersDnd) ? ' id="'.(((int)(Tools::getValue($this->identifiersDnd[$this->identifier], 1))) ? substr($this->identifier,3,strlen($this->identifier)) : '').'"' : '' ).' class="table'.((array_key_exists($this->identifier, $this->identifiersDnd) && ($this->_orderBy != 'position '&& $this->_orderWay != 'DESC')) ? ' tableDnD'  : '' ).'" cellpadding="0" cellspacing="0">
```
Comment (jeromenadaud) [inline classes/AdminTab.php]: Hi, Maybe use Tools::substr and Tools::strlen, and space after $this->_orderBy != 'position '

File: classes/AdminTab.php [hunk 6]
```diff
@@ -2081,11 +2075,11 @@ public function displayOptionTypeSelectLang($key, $field, $value)
...
-			echo  '<select name="'.$key.'_'.strtoupper($language['iso_code']).'">';
+			echo '<select name="'.$key.'_'.strtoupper($language['iso_code']).'">';
 			foreach ($field['list'] as $k => $v)
-				echo  '<option value="'.(isset($v['cast']) ? $v['cast']($v[$field['identifier']]) : $v[$field['identifier']]).'"'.((htmlentities(Tools::getValue($key.'_'.strtoupper($language['iso_code']), (Configuration::get($key.'_'.strtoupper($language['iso_code'])) ? Configuration::get($key.'_'.strtoupper($language['iso_code'])) : '')), ENT_COMPAT, 'UTF-8') == $v[$field['identifier']]) ? ' selected="selected"' : '').'>'.$v['name'].'</option>';
-			echo  '</select>';
-			echo  '</div>';
+				echo '<option value="'.(isset($v['cast']) ? $v['cast']($v[$field['identifier']]) : $v[$field['identifier']]).'"'.((htmlentities(Tools::getValue($key.'_'.strtoupper($language['iso_code']), (Configuration::get($key.'_'.strtoupper($language['iso_code'])) ? Configuration::get($key.'_'.strtoupper($language['iso_code'])) : '')), ENT_COMPAT, 'UTF-8') == $v[$field['identifier']]) ? ' selected="selected"' : '').'>'.$v['name'].'</option>';
```
Comment (jeromenadaud) [inline classes/AdminTab.php]: Hi, Why not use Tools::strtoupper() ?

Comment (PrestaEdit) [inline classes/AdminTab.php]: Done ! :)

**Comment by PrestaEdit** [comment]
Hi @jeromenadaud ,

Done for the asked change ! :)

File: classes/controller/FrontController.php [hunk 7]
```diff
@@ -333,15 +333,15 @@ public function init()
...
+			$cart->id_lang = (int)$this->context->cookie->id_lang;
+			$cart->id_currency = (int)$this->context->cookie->id_currency;
+			$cart->id_guest = (int)$this->context->cookie->id_guest;
 			$cart->id_shop_group = (int)$this->context->shop->id_shop_group;
 			$cart->id_shop = $this->context->shop->id;
 			if ($this->context->cookie->id_customer)
 			{
-				$cart->id_customer = (int)($this->context->cookie->id_customer);
-				$cart->id_address_delivery = (int)(Address::getFirstCustomerAddressId($cart->id_customer));
+				$cart->id_customer = (int)$this->context->cookie->id_customer);
```
Comment (jeromenadaud) [inline classes/controller/FrontController.php]: Hi, extra parenthesis at the end. ;-)

Comment (PrestaEdit) [inline classes/controller/FrontController.php]: Done ! :)

File: classes/order/Order.php [hunk 8]
```diff
@@ -814,17 +814,17 @@ public function hasProductReturned()
 
 	public function hasBeenPaid()
 	{
-		return count($this->getHistory((int)($this->id_lang), false, false, OrderState::FLAG_PAID));
+		return count($this->getHistory((int)$this->id_lang), false, false, OrderState::FLAG_PAID));
```
Comment (jeromenadaud) [inline classes/order/Order.php]: Hi, extra parenthesis after id_lang ;-)

Comment (PrestaEdit) [inline classes/order/Order.php]: Done ! :)

File: classes/webservice/WebserviceSpecificManagementImages.php [hunk 9]
```diff
@@ -857,27 +846,27 @@ protected function manageDeclinatedImagesCRUD($filename_exists, $filename, $imag
...
 	 *
-	 * @param string $filePath the image file path
+	 * @param string $file_path the image file path
 	 * @param array $imageTypes The differents sizes
-	 * @param string $parentPath The parent path
+	 * @param string $parent_path The parent path
 	 * @return boolean
 	 */
-	protected function deleteImageOnDisk($filePath, $imageTypes = null, $parentPath = null)
+	protected function deleteImageOnDisk($file_path, $imageTypes = null, $parent_path = null)
```
Comment (jeromenadaud) [inline classes/webservice/WebserviceSpecificManagementImages.php]: Hi, why not change also imageTypes for image_types ?

File: classes/webservice/WebserviceSpecificManagementImages.php [hunk 10]
```diff
@@ -897,100 +886,100 @@ protected function deleteImageOnDisk($filePath, $imageTypes = null, $parentPath
...
+	 * @param string $new_path
+	 * @param int $dest_width
+	 * @param int $dest_height
 	 * @param array $imageTypes
-	 * @param string $parentPath
+	 * @param string $parent_path
 	 * @return string
 	 */
-	protected function writeImageOnDisk($basePath, $newPath, $destWidth = null, $destHeight = null, $imageTypes = null, $parentPath = null)
+	protected function writeImageOnDisk($base_path, $new_path, $dest_width = null, $dest_height = null, $imageTypes = null, $parent_path = null)
```
Comment (jeromenadaud) [inline classes/webservice/WebserviceSpecificManagementImages.php]: Hi, the same here for imageType

File: classes/webservice/WebserviceSpecificManagementImages.php [hunk 11]
```diff
@@ -998,37 +987,35 @@ protected function writeImageOnDisk($basePath, $newPath, $destWidth = null, $des
...
+	 * @param string $reception_path
+	 * @param int $dest_width
+	 * @param int $dest_height
 	 * @param array $imageTypes
-	 * @param string $parentPath
+	 * @param string $parent_path
 	 * @return boolean
 	 */
-	protected function writePostedImageOnDisk($receptionPath, $destWidth = null, $destHeight = null, $imageTypes = null, $parentPath = null)
+	protected function writePostedImageOnDisk($reception_path, $dest_width = null, $dest_height = null, $imageTypes = null, $parent_path = null)
```
Comment (jeromenadaud) [inline classes/webservice/WebserviceSpecificManagementImages.php]: HI, imageTypes again ;-)

**Comment by jeromenadaud** [comment]
Hi @PrestaEdit  I'am ready to merge, just review last comments please.

Best regards.

**Comment by PrestaEdit** [comment]
Hi @jeromenadaud I just do it. :)

Best regards.

**Comment by jeromenadaud** [comment]
Hi, 
You've done it, great job !

Best regards.

**Comment by PrestaEdit** [comment]
:+1:

**Pull request merged**